src/Avalonia.Visuals/Media/Fonts/FamilyNameCollection.cs

```diff
@@ -54,30 +54,67 @@ public FamilyNameCollection(string familyNames)
         /// </value>
         internal IReadOnlyList<string> Names { get; }
 
-        /// <inheritdoc />
         /// <summary>
-        /// Returns an enumerator that iterates through the collection.
+        /// Returns an enumerator for the name collection.
         /// </summary>
-        /// <returns>
-        /// An enumerator that can be used to iterate through the collection.
-        /// </returns>
-        public IEnumerator<string> GetEnumerator()
+        public Enumerator GetEnumerator()
         {
-            return Names.GetEnumerator();
+            return new Enumerator(this);
```

Comment:
Interested to know why you implemented a custom enumerator here. Does this show up as being a hot path in profiling somewhere? I feels like enumerating font family names wouldn't be something done often enough to become a hot path.

Reply:
The FontNameCollection is part of the equality implementation and it is evaluated every time we request a typeface. So this is potentially called many times.